pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -489,6 +562,31 @@ private void checkCanRemovePendingAcksAndHandle(PositionImpl position, MessageId
         }
     }
 
+    private Consumer getAckOwnerConsumer(long ledgerId, long entryId) {
+        Consumer ackOwnerConsumer = this;
+        if (Subscription.isIndividualAckMode(subType)) {
+            for (Consumer consumer : subscription.getConsumers()) {
+                if (consumer != this && consumer.getPendingAcks().containsKey(ledgerId, entryId)) {
+                    ackOwnerConsumer = consumer;
+                    break;
+                }
+            }
+        }
+        return ackOwnerConsumer;
+    }
+
+    private boolean isDeletionAtBatchIndexLevelEnabled() {
+        ServiceConfiguration configuration = subscription.getTopic().getBrokerService().getPulsar().getConfiguration();
```

Comment:
is it necessary to always call this chain of methods ?
what about caching this value when we create this object ?

Reply:
Thanks, I have put it in the constructor method.